File: turborepo/packages/common/model/dto/dashboard-dto.ts
export namespace DashboardDto {
  export interface Stats {
    totalAccounts: number
    activeAccounts: number
    weeklyTrend: DailySignup[]
    statusCounts: StatusCount[]
    recentAccounts: RecentAccount[]
  }

  export interface DailySignup {
    date: string   // "2026-03-24"
    count: number
  }

  export interface StatusCount {
    status: string
    count: number
  }

  export interface RecentAccount {
    no: number
    id: string
    name?: string
    email?: string
    roles?: string[]
    status?: string
    createdAt: string
  }
}

File: backend/api/src/main/java/com/system/alba/model/dto/DashboardDto.java
package com.system.alba.model.dto;

import com.system.alba.common.AppType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

public class DashboardDto {

    @Getter
    @Setter
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Stats {
        private long totalAccounts;
        private long activeAccounts;
        private List<DailySignup> weeklyTrend;
        private List<StatusCount> statusCounts;
        private List<RecentAccount> recentAccounts;
    }

    @Getter
    @Setter
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class DailySignup {
        private String date;   // "2026-03-24"
        private long count;
    }

    @Getter
    @Setter
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class StatusCount {
        private String status;
        private long count;
    }

    @Getter
    @Setter
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class RecentAccount {
        private Long no;
        private String id;
        private String name;
        private String email;
        private List<AppType.AccountRole> roles;
        private AppType.Status status;
        private LocalDateTime createdAt;
    }
}
